src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -0,0 +1,571 @@
+                            </template>
+                          </v-select>
+                        </v-col>
+                      </v-row>
+                    </template>
+                  </v-radio>
+                  <v-radio :value="true" label="No par value" id="radio-no-par"/>
+                </v-radio-group>
+
+                <div v-show="isSeries" class="pl-10">
+                  <v-row v-if="shareStructure.hasParValue">
+                    <v-col cols="6">
+                      <v-text-field
+                        label="Par Value"
+                        id="series-par-value"
+                        :value="shareStructure.parValue"
+                        :disabled="true"
+                        width="10"/>
+                    </v-col>
+                    <v-col cols="6">
+                      <v-text-field
+                        id="series-currency"
+                        label="Currency"
+                        :value="`${getCurrencyNameByCode(shareStructure.currency)} (${shareStructure.currency})`"
+                        :disabled="true"/>
+                    </v-col>
+                  </v-row>
+                  <v-label id="lbl-no-par" v-else>No par value</v-label>
+                </div>
+
+                <v-divider class="separator mx-4" />
+
+                <div class="form__row">
+                  <v-checkbox
+                    id="special-rights-check-box"
+                    :label="'This share ' + shareStructure.type.toLowerCase() + ' has special rights or restrictions'"
+                    v-model="shareStructure.hasRightsOrRestrictions"
+                    @click="confirmSeriesRemoval()"
+                  />
+                </div>
+
+                <div class="form__row form__btns">
+                  <v-btn large outlined color="error" :disabled="activeIndex === -1"
+                         @click="removeShareStructure()" id="remove-btn">Remove</v-btn>
+
+                  <v-btn
+                    large
+                    id="done-btn"
+                    class="form-primary-btn"
+                    color="primary"
+                    @click="validateForm()"
+                  >
+                    Done
+                  </v-btn>
+
+                  <v-btn id="cancel-btn" large outlined color="primary" @click="resetFormAndData(true)">
+                    Cancel
+                  </v-btn>
+                </div>
+              </v-form>
+            </div>
+          </div>
+        </li>
+      </ul>
+    </v-expand-transition>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Emit, Mixins, Prop, Vue, Watch } from 'vue-property-decorator'
+
+// Components
+import { ConfirmDialog } from '@/components/ConfirmDialog'
+
+// Interfaces and Enums
+import { ConfirmDialogType, FormIF, ShareClassIF } from '@/interfaces'
+
+// Mixins
+import { CurrencyLookupMixin } from '@/mixins'
+import { ActionTypes } from '@/enums'
```

Comment:
As above, I think this should import from the shared components (enums and interfaces).

What about mixins -- are those a shared component, or does this component expect the app to provide the necessary mixins?

Reply:
If this component expects the app to provide the CurrencyLookupMixin, I think we should start adding comments to that effect (eg, line 176). Maybe...

`// this component expects this mixin/interface/enum/blah to exist in the app`

or something to the effect that it's not importing its own dependencies from the shared library.

Reply:
Good point, i think if we were relying on something the app would be providing.
Is the comment needed when the import is contained within the shared repo?
Doesn't the import already imply that?

Reply:
> Good point, i think if we were relying on something the app would be providing.
👍 

> Is the comment needed when the import is contained within the shared repo?
> Doesn't the import already imply that?

Right. I think it's obvious then. Only when the shared component expects the app to provide the local thing.